tests/cnf/core/network/metallb/tests/bgp-unnumbered.go

```diff
@@ -0,0 +1,282 @@
+					tsparams.ExternalMacVlanNADName, tsparams.TestNamespaceName, srIovInterfacesUnderTest[0],
+					ethernetIPAddresses)
+
+				frrPod := createFrrPod(
+					workerNodeList[1].Object.Name, frrConfigMap.Definition.Name, []string{}, frrStaticIPAnnotation)
+
+				By("Creating BGP Peers")
+				createBGPPeerUnnumberedAndVerifyIfItsReady(tsparams.BgpPeerName1, tsparams.BgpPeerDynamicASiBGP,
+					srIovInterfacesUnderTest[0], bfdProfileName, tsparams.LocalBGPASN, 0, false,
+					0, frrk8sPods)
+
+				By("Checking that BGP session is established and up")
+				verifyMetalLbBGPSessionsAreUPOnFrrPod(frrPod, []string{srIovInterfacesUnderTest[0]})
+			})
+
+		It("Verify eBGP peering established and advertised IPv4 prefixes received by the external FRR pod",
+			reportxml.ID("80394"), func() {
+				By("Verifying that the frrk8sPod deployment is in Ready state and create a list of the pods on " +
+					"worker nodes.")
+				By("Creating nginx test pod on worker node 0")
+				setupNGNXPod(workerNodeList[0].Definition.Name, tsparams.LabelValue1)
+
+				By("Creating static ip annotation for the external FRR pod")
+				frrStaticIPAnnotation := pod.StaticIPAnnotationWithInterfaceAndNamespace(
+					tsparams.ExternalMacVlanNADName, tsparams.TestNamespaceName, srIovInterfacesUnderTest[0],
+					ethernetIPAddresses)
+
+				By(fmt.Sprintf("Verify the frrk8s pod's link local IP address from interface %s on worker node %s",
+					srIovInterfacesUnderTest[0], workerNodeList[0].Definition.Name))
+				linkLocalAddress := getLinkLocalAddress(frrk8sPods[0], srIovInterfacesUnderTest[0], tsparams.FRRContainerName)
+
+				By("Create a frr config-map")
+				frrConfigMap := createConfigMapWithUnnumbered(tsparams.RemoteBGPASN, tsparams.LocalBGPASN,
+					srIovInterfacesUnderTest[0], linkLocalAddress, externalAdvertisedIPv4Routes, externalAdvertisedIPv6Routes,
+					false, true)
+
+				By("Creating BGP Peers")
+				createBGPPeerUnnumberedAndVerifyIfItsReady(tsparams.BgpPeerName1, tsparams.BgpPeerDynamicASeBGP,
+					srIovInterfacesUnderTest[0], bfdProfileName, tsparams.LocalBGPASN, 0, false,
+					0, frrk8sPods)
+
+				frrPod := createFrrPod(
+					workerNodeList[1].Object.Name, frrConfigMap.Definition.Name, []string{}, frrStaticIPAnnotation)
+
+				By("Checking that BGP session is established and up")
+				verifyMetalLbBGPSessionsAreUPOnFrrPod(frrPod, []string{srIovInterfacesUnderTest[0]})
+
+				By("Validating BGP route prefix")
+				validatePrefix(frrPod, netparam.IPV4Family, netparam.IPSubnetInt32, []string{linkLocalAddress},
+					ipAddressPoolRange)
+			})
+
+		It("Verify iBGP peering established with configured peerASN and unnumbered",
+			reportxml.ID("80395"), func() {
+				By("Verifying that the frrk8sPod deployment is in Ready state and create a list of the pods on " +
+					"worker nodes.")
+				By("Creating nginx test pod on worker node 0")
+				setupNGNXPod(workerNodeList[0].Definition.Name, tsparams.LabelValue1)
```

Comment:
do we need nginx in the It?

Reply:
The nginx pods are in the test namespace. The test namespace is cleaning all pods between the test cases so yes.

Reply:
let me ask differently , if we remove it which assertion in the spec will fail?